contrib/babelfishpg_tds/src/backend/tds/tdsresponse.c

```diff
@@ -1725,11 +1725,15 @@ PrepareRowDescription(TupleDesc typeinfo, List *targetlist, int16 *formats,
 												att->attcollation, (atttypmod - 4) * 2);
 				break;
 			case TDS_SEND_VARCHAR:
+				if (atttypmod == TSQLMaxTypmod)
+					atttypmod = -1;
 				SetColMetadataForCharTypeHelper(col, TDS_TYPE_VARCHAR,
 												att->attcollation, (atttypmod == -1) ?
 												atttypmod : (atttypmod - 4));
 				break;
 			case TDS_SEND_NVARCHAR:
```

Comment:
Looks like this should be common issue with: varbinarytypmodin, bpchartypmodin and varchartypmodin
but you have only fixed n/varchar